sdmetrics/reports/single_table/diagnostic_report.py

```diff
@@ -1,75 +1,83 @@
+from sdmetrics.reports.single_table._properties import Boundary, Coverage, Synthesis
+from sdmetrics.reports.single_table.base_report import BaseReport
+from sdmetrics.reports.utils import DIAGNOSTIC_REPORT_RESULT_DETAILS, print_results_for_level
 
 LOGGER = logging.getLogger(__name__)
 
 
-class DiagnosticReport():
+class DiagnosticReport(BaseReport):
     """Single table diagnostic report.
 
     This class creates a diagnostic report for single-table data. It calculates the diagnostic
-    score along three properties - synthesis, coverage, and boundaries.
+    score along three properties - Synthesis, Coverage, and Boundary.
     """
 
-    METRICS = {
-        'Coverage': [RangeCoverage, CategoryCoverage],
-        'Synthesis': [NewRowSynthesis],
-        'Boundaries': [BoundaryAdherence],
-    }
-    _METRIC_ARGS = {
-        'NewRowSynthesis': {'synthetic_sample_size': 10000},
-    }
-
     def __init__(self):
-        self._metric_results = {}
-        self._metric_averages = {}
-        self._property_scores = {}
-        self._results = {}
-        self._metric_args = copy.deepcopy(self._METRIC_ARGS)
+        super().__init__()
+        self._properties = {
+            'Coverage': Coverage(),
+            'Boundary': Boundary(),
+            'Synthesis': Synthesis()
+        }
+        self.results = {}
+
+    def _generate_results(self):
+        """Generate the diagnostic report results."""
+        if not self.results:
+            self.results['SUCCESS'] = []
+            self.results['WARNING'] = []
+            self.results['DANGER'] = []
+
+            for property_name in self._properties:
+                details = self._properties[property_name]._details
+                average_score_metric = details.groupby('Metric')['Score'].mean()
+                for metric, score in average_score_metric.items():
+                    if pd.isna(score):
+                        continue
+                    if score >= 0.9:
+                        self.results['SUCCESS'].append(
+                            DIAGNOSTIC_REPORT_RESULT_DETAILS[metric]['SUCCESS'])
+                    elif score >= 0.5:
+                        self.results['WARNING'].append(
+                            DIAGNOSTIC_REPORT_RESULT_DETAILS[metric]['WARNING'])
+                    else:
+                        self.results['DANGER'].append(
+                            DIAGNOSTIC_REPORT_RESULT_DETAILS[metric]['DANGER'])
+
+    def _get_num_iterations(self, property_name, metadata):
+        """Get the number of iterations for the property.
+
+        Args:
+            property_name (str):
+                The name of the property.
+            metadata (dict):
+                The metadata of the table.
+        """
+        self._check_property_name(property_name)
+
+        num_columns = len(metadata['columns'])
+
+        if property_name in ['Coverage', 'Boundary']:
+            return num_columns
+        elif property_name == 'Synthesis':
```

Comment:
would use a blank line before this `elif` to see better the return

Reply:
Yes, done in [fcae5dc](https://github.com/sdv-dev/SDMetrics/pull/411/commits/fcae5dc0a63db277f9b06746c587f2fdd0fb2cd0).